1DIP/deriv2_cgls.ipynb

```diff
@@ -186,229 +264,249 @@
-   "execution_count": null,
-   "id": "1a523f4b",
-   "metadata": {},
-   "outputs": [],
-   "source": []
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 93,
-   "id": "a8bd2985",
+   "execution_count": 10,
+   "id": "28589f81",
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "     Iter   Max Iter     Time/Iter            Objective\n",
-      "                               [s]                     \n",
-      "        0          4         0.000          2.09949e-03\n",
-      "        1          4         0.190          3.28010e-05\n",
-      "        2          4         0.199          3.27442e-06\n",
-      "        3          4         0.184          1.38428e-06\n",
-      "        4          4         0.176          1.08585e-06\n",
-      "-------------------------------------------------------\n",
-      "        4          4         0.176          1.08585e-06\n",
-      "Stop criterion has been reached.\n",
-      "\n"
+      "[ 5.94142791e-17 -8.89045781e-18  1.18828558e-16 -1.47017815e-16\n",
+      "  3.64291930e-16 -8.69096461e-16  1.32185929e-15 -1.61849700e-15\n",
+      "  1.70349845e-15 -6.80011603e-16 -6.12357387e-16  1.57165947e-15\n",
+      " -1.10328413e-15 -3.81639165e-16  1.43288159e-15 -9.29811783e-16\n",
+      " -5.68989300e-16  1.21430643e-15 -2.77555756e-17 -1.33920652e-15\n",
+      "  7.28583860e-16  1.20042865e-15 -1.31145095e-15 -3.12250226e-16\n",
+      "  1.49186219e-15  3.95516953e-16 -1.22124533e-15  2.77555756e-17\n",
+      "  1.51267887e-15  6.93889390e-17 -1.40859546e-15  2.28983499e-16\n",
+      "  1.34614542e-15 -6.24500451e-17 -1.30451205e-15  1.31838984e-16\n",
+      "  1.28369537e-15  4.57966998e-16 -9.43689571e-16 -5.34294831e-16\n",
+      "  1.21430643e-15  6.73072709e-16 -6.80011603e-16 -9.57567359e-16\n",
+      "  3.33066907e-16  1.31838984e-15 -1.31838984e-16 -1.22124533e-15\n",
+      " -5.34294831e-16  9.43689571e-16  7.63278329e-16 -8.88178420e-16\n",
+      " -1.02695630e-15  6.93889390e-17  8.18789481e-16  2.77555756e-17\n",
+      " -9.43689571e-16 -9.57567359e-16  5.96744876e-16  9.57567359e-16\n",
+      " -1.38777878e-16 -1.17961196e-15 -7.21644966e-16  5.96744876e-16\n",
+      "  9.02056208e-16 -5.55111512e-16 -1.35308431e-15 -3.88578059e-16\n",
+      "  8.46545056e-16  7.91033905e-16 -5.06539255e-16 -1.15879528e-15\n",
+      " -5.41233725e-16  9.02056208e-16  6.59194921e-16 -5.27355937e-16\n",
+      " -1.23512311e-15 -4.30211422e-16  8.88178420e-16  9.15933995e-16\n",
+      " -4.37150316e-16 -1.20736754e-15 -5.68989300e-16  8.04911693e-16\n",
+      "  9.92261828e-16 -2.08166817e-16 -1.11022302e-15 -6.10622664e-16\n",
+      "  7.49400542e-16  1.08246745e-15 -1.38777878e-17 -1.01654796e-15\n",
+      " -7.59808882e-16  6.14092110e-16  1.10328413e-15  2.18575158e-16\n",
+      " -9.87057658e-16 -8.69096461e-16  4.18068358e-16  1.17354043e-15]\n"
+     ]
+    },
+    {
+     "ename": "TypeError",
```

Comment:
There is an error showing up in the output here - needs to be resolved.

Reply:
Thanks - done!